packages/common/src/failure.ts

```diff
@@ -329,15 +329,22 @@ export async function errorToFailure(err: unknown, dataConverter: DataConverter)
     source: FAILURE_SOURCE,
   };
 
-  if (err instanceof Error) {
-    return { ...base, message: err.message ?? '', stackTrace: cutoffStackTrace(err.stack) };
+  if (isRecord(err) && hasOwnProperties(err, ['message', 'stack'])) {
+    return {
+      ...base,
+      message: String(err.message) ?? '',
+      stackTrace: cutoffStackTrace(String(err.stack)),
+      cause: await optionalErrorToOptionalFailure(err.cause, dataConverter),
+    };
   }
 
+  const stackTrace = `A non-Error value was thrown from your code. We recommend throwing Error objects so that we can provide a stack trace.`;
```

Comment:
Not sure I'd use this as the stack trace because it's not formatted like one, consider putting this in the message.